perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java

```diff
@@ -159,6 +194,31 @@ public Application approveApplication(PerunSession session, Application app) thr
 
 		}
 
+		// create ues RC for future merging
+		List<ApplicationFormItemData> data = new ArrayList<>();
+		try {
+			data = registrar.getApplicationDataById(session, app.getId());
+		} catch (RegistrarException e) {
+			// ignore because application's id is not null
+		}
+		for (ApplicationFormItemData item : data) {
+			if (item.getFormItem() != null &&
+				Objects.equals(AttributesManager.NS_USER_ATTR_DEF + ":birthNumber", item.getFormItem().getPerunDestinationAttribute())) {
+
+				String rc = item.getValue();
+				if (rc != null && !rc.isEmpty()) {
+					ExtSource es = perun.getExtSourcesManager().checkOrCreateExtSource(session, "RC", rc);
+					UserExtSource ues = new UserExtSource(es, app.getExtSourceLoa(), app.getCreatedBy());
```

Comment:
You shouldn't pass identity from application into the RC UserExtSource. It should contain login from the rc, like 
```java
new UserExtSource(es, 0, rc));
```

Reply:
Fixed.